app/models/concerns/hud_concerns/enrollment.rb

```diff
@@ -49,7 +49,9 @@ def self.invalidate_processing!
 
     def invalidate_processing!
       # use update_columns to avoid stale object errors when this method is called from Service after_commit hooks
-      update_columns(processed_as: nil, processed_hash: nil)
+      self.class.without_optimistic_locking do
```

Comment:
This feels right here.  The invalidate processing shouldn't fail just because the record is stale.  Thanks.